common/tools/field_test_logger/wifi_info.py

```diff
@@ -23,6 +24,14 @@ def __init__(self, interval, interface, batman_interface):
 
     # ----------------------------------------
 
+    def __run_command(self, cmd):
+        try:
+            with subprocess.Popen(cmd, stdout=subprocess.PIPE) as proc:
+                return proc.communicate()[0].decode().rstrip()
```

Comment:
Use `subprocess.run(cmd, text=True).stdout` instead (you might also want to capture stderr with `stderr=subprocess.STDOUT`), or use `subprocess.check_output(cmd, text=True)` and capture the `CalledProcessError` exception if the command shouldn't fail :construction:

Reply:
Yes, there is a lot of things to re-write.   Now we are trying to keep this process alive as a first 🩹